### DISCUSSION TITLE
Embed instead of link raster images

### DISCUSSION THREAD

**Discussion by robin-gdwl** [SVG-Edit/svgedit#749] [New feature request]
Hey everyone, I am using svgedit in my workshops with school-aged children. We are using it to create files that will be laser cut and engraved. 
I would love it if there was a possibility to embed a raster image directly into the SVG file instead of merely linking to an image and noting position and scale. This would make using the file in other programs much easier. 

What would it take to implement this? is it even a desirable feature for others using this tool?

**Comment by jfhenon** [comment]
Hello,
Nice to see your usage of svgedit. 

Yes, it's a desirable feature. 

In fact, you have already a way to imbed images instead of linking them.
You need to go in the document preferences (before you import the image) to select the option "imbed data"
<img width="271" alt="image" src="https://user-images.githubusercontent.com/20402845/161439255-65ee17ff-e882-4366-9d40-0f538c49fa0d.png">
I tested it and I found a few issues but it worked.
- you can only link a file in the image folder
- It's not working well with large embedded file (as svgedit will load everything in the browser memory).

This feature should be improved to 
- ask at the import time if you want to imbed or link the file
- allow to browse a file on your local system 

You could open an issue for this but I can't commit it will be implemented and when.
